packages/worker/src/connection.ts

```diff
@@ -10,18 +11,56 @@ import { Runtime } from './runtime';
  * Do not confuse this connection class with `@temporalio/client`'s Connection.
  */
 export class NativeConnection {
-  // nativeClient is intentionally left private, framework code can access it with `extractNativeClient` (below)
-  private constructor(private nativeClient: Client) {}
+  /**
+   * referenceHolders is used internally by the framework, it can be accessed with `extractReferenceHolders` (below)
+   */
+  private readonly referenceHolders = new Set<Worker>();
+
+  /**
+   * nativeClient is intentionally left private, framework code can access it with `extractNativeClient` (below)
+   */
+  protected constructor(private nativeClient: Client) {}
 
   static async create(options?: NativeConnectionOptions): Promise<NativeConnection> {
     const client = await Runtime.instance().createNativeClient(options);
     return new this(client);
   }
+
+  /**
+   * Close this connection.
+   *
+   * Make sure any Workers using this connection are stopped before calling
+   * this method or it will throw an {@link IllegalStateError}
+   */
+  async close(): Promise<void> {
+    if (this.referenceHolders.size > 0) {
+      throw new IllegalStateError('Cannot close connection while Workers hold a reference to it');
+    }
+    await Runtime.instance().closeNativeClient(this.nativeClient);
+  }
 }
 
 /**
- * Extract the private native client instance from a `NativeConnection` instance
+ * Extract the private native client instance from a `NativeConnection` instance.
+ *
+ * Only meant to be used by the framework.
```

Comment:
Is there a "hidden doc" annotation?

Reply:
No need, we never re-export this

Reply:
(Only things exported from package's `index.ts` show up in API reference)